### PR TITLE
BUG: MultiIndex.drop does not raise if labels are partially found

### LINKED ISSUE #37820: BUG: MultiIndex.drop does not raise if labels are partially found
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
mi = MultiIndex.from_tuples([(1, 2), (2, 2), (3, 2)])
mi.drop([1, 4], level=0) # does not raise
mi.drop(4, level=0) # raises


```

#### Problem description
The docs say
```
Raises

    KeyError

        If not all of the labels are found in the selected axis

```
I think both statements should raise ``KeyError``
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.drop.html

#### Expected Output

Both are raising ``KeyError: 'labels [4] not found in level'``

For regular Indexes this holds True

```
index = Index([1, 2])
index.drop([1, 3]) # Raises
```

#### Output of ``pd.show_versions()``

<details>

master

</details>

### PR DESCRIPTION
**Pull request by GYHHAHA**
- [x] closes #37820
- [x] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/multi.py [hunk 1]
```diff
@@ -2157,9 +2157,10 @@ def _drop_from_level(self, codes, level, errors="raise"):
         index = self.levels[i]
         values = index.get_indexer(codes)
 
+        not_found = np.array(codes)[np.array(values) == -1].tolist()
```
Comment (phofl) [inline pandas/core/indexes/multi.py:2160]: That won't work after #37794 is merged. Could you rebase afterwards? Please check for len(not_found)

Comment (GYHHAHA) [reply pandas/core/indexes/multi.py:2160]: ok

Comment (jbrockmendel) [reply pandas/core/indexes/multi.py:2160]: is the to_list necessary?

Comment (GYHHAHA) [reply pandas/core/indexes/multi.py:2160]: not necessary, I will alter this after phofl's PR merged.

**Review by jreback — CHANGES_REQUESTED** [review]
will review after rebase

File: pandas/core/indexes/multi.py (hunk 1 above)
Comment (phofl) [reply pandas/core/indexes/multi.py:2160]: You could rebase now

**Comment by GYHHAHA** [comment]
cc @jreback

File: pandas/tests/indexes/multi/test_drop.py [hunk 2]
```diff
@@ -147,3 +147,21 @@ def test_drop_with_nan_in_index(nulls_fixture):
...
+    mi = MultiIndex.from_tuples([(1, 2), (2, 2), (3, 2)])
+    msg = r"labels \[4\] not found in level"
+    with pytest.raises(KeyError, match=msg):
+        mi.drop(4, level=0)
+    with pytest.raises(KeyError, match=msg):
+        mi.drop([1, 4], level=0)
+    msg = r"labels \[nan\] not found in level"
+    with pytest.raises(KeyError, match=msg):
+        mi.drop([np.nan], level=0)
+    mi = MultiIndex.from_tuples([(np.nan, 1), (1, 2)])
```
Comment (jreback) [inline pandas/tests/indexes/multi/test_drop.py:164]: can you add a blank line here  (before you create the mi, L164)

**Review by jreback — CHANGES_REQUESTED** [review]
looks good, very minor comment, can you merge master and ping on green.

cc @phofl @jbrockmendel if comments

File: pandas/tests/indexes/multi/test_drop.py (hunk 2 above)
Comment (GYHHAHA) [reply pandas/tests/indexes/multi/test_drop.py:165]: done

File: pandas/tests/indexes/multi/test_drop.py [hunk 3]
```diff
@@ -147,3 +147,22 @@ def test_drop_with_nan_in_index(nulls_fixture):
     msg = r"labels \[Timestamp\('2001-01-01 00:00:00'\)\] not found in level"
     with pytest.raises(KeyError, match=msg):
         mi.drop(pd.Timestamp("2001"), level="date")
+
+
+def test_single_level_drop():
```
Comment (phofl) [inline pandas/tests/indexes/multi/test_drop.py:152]: Could you rename to something like test_single_level_drop_partially_missing_elements?

File: pandas/tests/indexes/multi/test_drop.py [hunk 4]
```diff
@@ -147,3 +147,22 @@ def test_drop_with_nan_in_index(nulls_fixture):
...
+    mi = MultiIndex.from_tuples([(1, 2), (2, 2), (3, 2)])
+    msg = r"labels \[4\] not found in level"
+    with pytest.raises(KeyError, match=msg):
+        mi.drop(4, level=0)
+    with pytest.raises(KeyError, match=msg):
+        mi.drop([1, 4], level=0)
+    msg = r"labels \[nan\] not found in level"
+    with pytest.raises(KeyError, match=msg):
+        mi.drop([np.nan], level=0)
+
```
Comment (phofl) [inline pandas/tests/indexes/multi/test_drop.py:164]: Could you add a case, where a list of elements should be dropped where nan is the only missing element? This is a bit of an edge case after our changes, but should work as far as I can see

**Review by phofl — COMMENTED** [review]
Thx, minor comments

File: pandas/tests/indexes/multi/test_drop.py (hunk 4 above)
Comment (GYHHAHA) [reply pandas/tests/indexes/multi/test_drop.py:166]: done

File: pandas/tests/indexes/multi/test_drop.py (hunk 3 above)
Comment (jreback) [reply pandas/tests/indexes/multi/test_drop.py:152]: @GYHHAHA can you do this one, ping on green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @GYHHAHA very nice, keep em coming!

**Pull request merged**